### PR TITLE
Fix data_layers and data_type function in topology.py

### PR DESCRIPTION
**Pull request by qingqing01**

### CODE REVIEW

File: python/paddle/v2/topology.py [hunk 1]
```diff
@@ -86,8 +91,12 @@ def data_type(self):
...
         [('image', dense_vector(768)), ('label', integer_value(10))]
         """
-        return [(data_layer.name, data_layer.type)
-                for data_layer in self.data_layers()]
+
+        data_types_lists = []
+        for each in self.__model_config__.input_layer_names:
+            data_layers = self.data_layers()
+            data_types_lists.append((each, data_layers[each].type))
+        return data_types_lists
```
Comment (jacquesqiao) [inline python/paddle/v2/topology.py]: 单测应该也需要修改一下

File: python/paddle/v2/topology.py [hunk 2]
```diff
@@ -86,8 +91,12 @@ def data_type(self):
         get data_type from proto, such as:
         [('image', dense_vector(768)), ('label', integer_value(10))]
         """
-        return [(data_layer.name, data_layer.type)
-                for data_layer in self.data_layers()]
+
+        data_types_lists = []
+        for each in self.__model_config__.input_layer_names:
+            data_layers = self.data_layers()
```
Comment (hedaoyuan) [inline python/paddle/v2/topology.py]: 这个可以提到for前面去吧。

**Comment by reyoung** [comment]
It seems that this issue was already fixed in #1465 in a better way?

**Comment by qingqing01** [comment]
@reyoung  Thanks!   Closed because it was fixed in #1465 .

**Pull request closed without merging**